### PR TITLE
Allow replay of past events for msal-angular

### LINKED ISSUE #4920: MSAL Angular Forgotten password error event not caught with PathLocationStrategy
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.26.0

### Wrapper Library

MSAL Angular (@azure/msal-angular)

### Wrapper Library Version

2.3.2

### Description

Hi, I have a problem and i'm unsure whether it is a bug or me doing something dumb.

I have reproduced the problem using the Angular 11 B2C sample application, with PathLocationStrategy routing enabled.
If I navigate to a **protected** route, I get redirected to the login screen, but if i click the forgotten password link, the LOGIN_FAILURE event is not raised back in my application (currently inside app.component.ts). 
However, if I click the login button from an **unprotected** page and click the forgotten password link, the event is raised as expected.

Also if I switch to using HashLocationStrategy then both scenarios work perfectly.

Below is the code i added to handle the AADB2C90118 code received for the forgotten password flow in app.component.ts.

```
this.msalBroadcastService.msalSubject$
      .pipe(
        filter(
          (msg: EventMessage) =>
            msg.eventType === EventType.LOGIN_FAILURE ||
            msg.eventType === EventType.ACQUIRE_TOKEN_FAILURE
        ), takeUntil(this._destroying$)
      ).subscribe((result: EventMessage) => {
        if (result.error) { 
          // Check for forgot password error
          if (result.error.message.includes('AADB2C90118')) {
            let resetPasswordFlowRequest = {
              scopes: ['openid'],
              authority: b2cPolicies.authorities.forgotPassword.authority,
            };
            this.login(resetPasswordFlowRequest);
          }
        }
      });
```

Many thanks for your help

### MSAL Configuration

```javascript
auth: {
      clientId: CLIENT_ID, // PPE testing environment
      authority: b2cPolicies.authorities.signUpSignIn.authority, // PPE testing environment.
      redirectUri: 'https://localhost:5001/auth',
      postLogoutRedirectUri: 'https://localhost:5001/',
      knownAuthorities: [b2cPolicies.authorityDomain],
    },
    cache: {
      cacheLocation: BrowserCacheLocation.LocalStorage,
      storeAuthStateInCookie: isIE, // set to true for IE 11
    },
    system: {
      loggerOptions: {
        loggerCallback,
        logLevel: LogLevel.Info,
        piiLoggingEnabled: false,
      },
    },
```

### Relevant Code Snippets

_No response_

### Identity Provider

Azure B2C Basic Policy

### Source

External (Customer)

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds optional configurations for the `MsalBroadcastService` to enable the replay of past events when subscribing to the `msalSubject$`. Includes unit tests, docs, and FAQ.

This addresses issue #4920 .

### CODE REVIEW

**Review by peterzenz — APPROVED** [review]

File: change/@azure-msal-angular-a7f4662d-701e-4ffc-bc3d-63067281ab2c.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (jasonnutter) [inline change/@azure-msal-angular-a7f4662d-701e-4ffc-bc3d-63067281ab2c.json]: I would mark this as minor given new functionality.

File: lib/msal-angular/src/msal.broadcast.service.ts [hunk 2]
```diff
@@ -3,24 +3,34 @@
...
+import { BehaviorSubject, Observable, ReplaySubject, Subject } from "rxjs";
+import { MSAL_BROADCAST_CONFIG, MSAL_INSTANCE } from "./constants";
 import { EventMessage, EventMessageUtils, IPublicClientApplication, InteractionStatus } from "@azure/msal-browser";
 import { MsalService } from "./msal.service";
+import { MsalBroadcastConfiguration } from "./msal.broadcast.config";
 
 @Injectable()
 export class MsalBroadcastService {
-    private _msalSubject: Subject<EventMessage>;
+    private _msalSubject: Subject<EventMessage>|ReplaySubject<EventMessage>;
```
Comment (jasonnutter) [inline lib/msal-angular/src/msal.broadcast.service.ts]: `ReplaySubject` subclasses `Subject`, so I think you can leave this as-is.

**Comment by jasonnutter** [comment]
A couple minor comments, otherwise looks good...great tests and docs!

File: lib/msal-angular/src/msal.broadcast.config.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export type MsalBroadcastConfiguration = {
+    replayPastEvents: number
```
Comment (jasonnutter) [inline lib/msal-angular/src/msal.broadcast.config.ts]: The name sounds like it is a boolean, I'd go for something like `eventsToReplay`

Comment (jo-arroyo) [reply lib/msal-angular/src/msal.broadcast.config.ts]: Makes sense. Changed.

File: lib/msal-angular/src/msal.broadcast.service.ts (hunk 2 above)
Comment (jo-arroyo) [reply lib/msal-angular/src/msal.broadcast.service.ts]: Got it. Changed.

File: change/@azure-msal-angular-a7f4662d-701e-4ffc-bc3d-63067281ab2c.json (hunk 1 above)
Comment (jo-arroyo) [reply change/@azure-msal-angular-a7f4662d-701e-4ffc-bc3d-63067281ab2c.json]: Done.

File: lib/msal-angular/src/msal.broadcast.service.spec.ts [hunk 4]
```diff
@@ -59,6 +66,238 @@ describe('MsalBroadcastService', () => {
...
+  });
+
+  it('does not broadcasts previous events if MsalBroadcastConfig is not set', (done) => {
+    const expectedMsalSubjectFirstSubscription = [
+      {
+        eventType: EventType.LOGIN_START,
+        interactionType: InteractionType.Redirect,
+        payload: null,
+        error: null,
+      },
```
Comment (Robbie-Microsoft) [inline lib/msal-angular/src/msal.broadcast.service.spec.ts]: super nitpick: I've noticed that you have inconsistently used trailing commas all throughout this review.

**Review by Robbie-Microsoft — APPROVED** [review]

File: lib/msal-angular/src/msal.broadcast.service.spec.ts (hunk 4 above)
Comment (jo-arroyo) [reply lib/msal-angular/src/msal.broadcast.service.spec.ts]: Thanks for the review and for pointing this out! Fixed the trailing commas.

**Pull request merged**